src/query/functions/temporal/aggregation.go

```diff
@@ -78,6 +82,30 @@ func (a aggProcessor) Init(op baseOp, controller *transform.Controller, opts tra
 	}
 }
 
+// NewQuantileOp create a new base temporal transform for quantile_over_time func
+func NewQuantileOp(args []interface{}, optype string) (transform.Params, error) {
+	if optype == QuantileType {
+		if len(args) != 2 {
+			return emptyOp, fmt.Errorf("invalid number of args for %s: %d", QuantileType, len(args))
+		}
+
+		scalar, ok := args[0].(float64)
+		if !ok {
+			return emptyOp, fmt.Errorf("unable to cast to scalar argument: %v for %s", args[1], QuantileType)
+		}
+
+		aggregationFunc := makeQuantileOverTimeFn(scalar)
```

Comment:
You can add an optimization here is `scalar < 0 || scalar > 1` by instead having a function that looks like:

```
func makeInvalidQuantileFn(inf float64) aggFunc {
 return func(_ []float64) float64 {
  return inf
 }
}
```
And passing the correctly signed infinity in